test/it/oauth_authenticator_it.js

```diff
@@ -154,6 +154,24 @@ describe('OAuthAuthenticator',function(){
     });
   });
 
+  it('should error when trying to authenticate with invalid stormpath token', function(done){
+    var authenticator = stormpath.OAuthAuthenticator(application);
+    authenticator.authenticate({
+      body: {
+        grant_type: 'stormpath_token',
+        stormpath_token: 'abc'
+      }
+    },function(err,result){
+      assert.isUndefined(result);
+
+      assert.isNotNull(err);
+      assert.equal(err.status, 400);
+      assert.equal(err.code, 10017); // 10017 = Token is invalid.
```

Comment:
Yeah, that would be a good idea. I think I'll look into doing a separate refactor for that.